### ISSUE TITLE
eslint no-unexpected-multiline error was ignored when I fixed with eslint

### ISSUE THREAD

**Issue by yinshuxun** [prettier/eslint-config-prettier#30]
[Playground link](https://prettier.io/playground/#%7B%22content%22%3A%22const%20a%20%3D%201%5Cn%20%20%20%20%20%20(()%20%3D%3E%20%7B%5Cn%20%20%20%20%20%20%20%20const%20c%20%3D%201%3B%5Cn%20%20%20%20%20%20%20%20console.log(c)%5Cn%20%20%20%20%20%20%7D)()%22%2C%22options%22%3A%7B%22ast%22%3Afalse%2C%22bracketSpacing%22%3Atrue%2C%22doc%22%3Afalse%2C%22jsxBracketSameLine%22%3Afalse%2C%22output2%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22printWidth%22%3A80%2C%22semi%22%3Atrue%2C%22singleQuote%22%3Afalse%2C%22tabWidth%22%3A2%2C%22trailingComma%22%3A%22none%22%2C%22useTabs%22%3Afalse%7D%7D)
```sh
# Options (if any):
--fix
```

**Input:**
```jsx
const a = 1
      (() => {
        const c = 1;
        console.log(c)
      })()
```

**Output:**
```jsx
const a = 1(() => {
  const c = 1;
  console.log(c);
})();
```

**Expected behavior:**

**Comment by lydell** [comment]
Hi! It is not clear what your issue is. Can you clarify?

**Comment by yinshuxun** [comment]
when I use eslint --fix , the code is form 
```
const a = 1
(() => {
    const c = 1;
    console.log(c);
})()
```
to
```
const a = 1(() => {
  const c = 1;
  console.log(c);
})();
```,
I think this should throw error，like the rule 'no-unexpected-multiline' in eslint

**Comment by yinshuxun** [comment]
@lydell you can see this [issue] in eslint (https://github.com/eslint/eslint/issues/9354)

**Comment by lydell** [comment]
Prettier takes valid JavaScript as input and produces valid JavaScript as output. That's it. It does not try to fix errors in your code.

This:

```js
const a = 1
(() => {
    const c = 1;
    console.log(c);
})()
```

... is valid JavaScript. Prettier doesn't use whitespace before the `(` paren of function calls, so Prettier turns it into:

```js
const a = 1(() => {
  const c = 1;
  console.log(c);
})();
```

If you run Prettier inside ESLint using [eslint-plugin-prettier](https://github.com/prettier/eslint-plugin-prettier) with the `--fix` flag, or if you just happen to run Prettier before ESLint (lots of people run Prettier on save in their editors), ESLint's `no-unexpected-multiline` rule doesn't get a chance to report the unexpected newline since it's already gone by that point.

This is unfortunate, and we haven't come up with any perfect solution. Here are some things you can do:

- Run Prettier (or `eslint --fix`) (on save) in your editor. Then you'll see the lines being merged together in real time and can just insert the missing semicolon.
- Adopt the habit of always adding a semicolon at the beginning of a line if it starts with `(` or `[`. (Prettier or ESLint can then (re)move it.) This is useful regardless of whether you are a "semicolon person" or "no semicolon person".
- Make sure to always run ESLint before Prettier.
- Test your code.
- Use Flow or TypeScript.

See also these prettier issues:

- https://github.com/prettier/prettier/issues/2662
- https://github.com/prettier/prettier/issues/1143
